### PR TITLE
[GPU] Enable asymmetric onednn convolution

### PR DESCRIPTION
**Pull request by jade-cho**
+ Use onednn convolution for asymmetric convolution when the weights zero points tensor is scalar.

### Tickets:
 - *142818*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.cpp [hunk 1]
```diff
@@ -120,7 +110,8 @@ struct convolution_onednn : typed_primitive_onednn_impl<convolution> {
 
     static std::shared_ptr<dnnl::primitive_attr> get_primitive_attributes(const typed_program_node<convolution>& arg,
                                                                             const kernel_impl_params& impl_params,
-                                                                            int& zero_point_mask) {
+                                                                            int& zero_point_mask,
+                                                                            dnnl::memory::data_type& wzp_data_type) {
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.cpp:114]: `wzp_data_type` is unused. Did you intend to set it from here?

File: src/plugins/intel_gpu/src/plugin/transformations/bcast_and_pad_zp_buffers.cpp [hunk 2]
```diff
@@ -106,8 +143,17 @@ BroadcastAndPadZeroPointBuffers::BroadcastAndPadZeroPointBuffers(size_t pad_size
 
         if (auto wzp = std::dynamic_pointer_cast<ov::op::v0::Constant>(pattern_map.at(wzp_m).get_node_shared_ptr())) {
             auto target_shape = wzp->get_shape();
-            const size_t wzp_c_idx = 0;
-            auto aligned_wzp = pad_quantization_parameter(wzp, out_shape[channel_idx].get_length(), wzp_c_idx, pad_size);
+
+            std::shared_ptr<ov::Node> aligned_wzp;
+            if (supports_immad && !(conv->get_groups() > 1)) {
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/plugin/transformations/bcast_and_pad_zp_buffers.cpp:148]: could you add a comment here?
`// OneDNN supports scalar wzp. If wzp data are identical, replace it with scalar value for OneDNN`

File: src/plugins/intel_gpu/tests/unit/test_cases/convolution_gpu_test.cpp [hunk 3]
```diff
@@ -9979,6 +9979,111 @@ TEST(convolution_gpu_onednn, quantized_onednn_convolution_u8s8f32_asymmetric_act
...
+
+                                   1, 2, -1,
+                                   -2, 1,  2,
+                                   9, 7, -1,
+
+                                   9, 0, -4,
+                                   -1, 3,  2,
+                                   0, 2,  5 });
+    set_values<uint8_t>(a_zp, { 2, 5, 5 });
+    set_values<uint8_t>(w_zp, { 0 });
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/unit/test_cases/convolution_gpu_test.cpp:10027]: could you use non-zero value for w_zp testing?

File: src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.cpp (hunk 1 above)
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.cpp:114]: It is used at create() function for binary serialization.

File: src/plugins/intel_gpu/src/plugin/transformations/bcast_and_pad_zp_buffers.cpp (hunk 2 above)
Comment (jade-cho) [reply src/plugins/intel_gpu/src/plugin/transformations/bcast_and_pad_zp_buffers.cpp:148]: Added

File: src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.cpp [hunk 4]
```diff
@@ -65,48 +69,34 @@ struct convolution_onednn : typed_primitive_onednn_impl<convolution> {
 
             GPU_DEBUG_GET_INSTANCE(debug_config);
             GPU_DEBUG_IF(debug_config->verbose >= static_cast<int>(ov::intel_gpu::LogLevel::TRACE_DETAIL)) {
```
[the remark this answers is not in the document]
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.cpp:71]: Will remove

File: src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.cpp [hunk 5]
```diff
@@ -65,48 +69,34 @@ struct convolution_onednn : typed_primitive_onednn_impl<convolution> {
...
-            //         GPU_DEBUG_TRACE_DETAIL << std::endl;
-            //         dnnl_mem.unmap_data(mapped_ptr);
-            //     }
-            // }
+            auto w_zp = instance.weights_zero_points_memory();
+            dnnl::memory::desc desc = onednn::layout_to_memory_desc(w_zp->get_layout(), dnnl::memory::format_tag::a, true);
+            args.insert({DNNL_ARG_ATTR_ZERO_POINTS | DNNL_ARG_WEIGHTS, w_zp->get_onednn_memory(desc)});
+
+            GPU_DEBUG_GET_INSTANCE(debug_config);
+            GPU_DEBUG_IF(debug_config->verbose >= static_cast<int>(ov::intel_gpu::LogLevel::TRACE_DETAIL)) {
```
[the remark this answers is not in the document]
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.cpp:83]: Will remove

File: src/plugins/intel_gpu/tests/unit/test_cases/convolution_gpu_test.cpp (hunk 3 above)
Comment (jade-cho) [reply src/plugins/intel_gpu/tests/unit/test_cases/convolution_gpu_test.cpp:10027]: Changed

**Review by isanghao — APPROVED** [review]
LGTM, no perf issue from daily test.

**Pull request merged**